src/librbd/api/Mirror.cc

```diff
@@ -548,12 +529,6 @@ int Mirror<I>::image_disable(I *ictx, bool force) {
       rollback = true;
       return r;
     }
-
-    r = ictx->operations->update_features(RBD_FEATURE_JOURNALING, false);
```

Comment:
Probably yes. But again, I am not sure it is a good idea (for my current implementation, where I assume that when mirroring is enabled we choose snapshot/journal mode based on if the journaling feature enabled). Because if one have an image with journaling enabled, then enable mirroring (journal mode) then disable, the feature is removed, enables again and now it is in snapshot mode. This might be confusing for the user.

Reply:
OK, how about you add a TODO and this could be part of a future PR to be explicit about the image mode.